### PR TITLE
Updated ops

### PR DESCRIPTION
**Pull request by aashokvardhan**
Updating the summary and introducing the 3d ops

## Release Notes

* **New Features**
  * Added 5 new 3D volumetric augmentation operators for medical imaging workflows
  * Added 6 new image processing operators including window normalization and grid masking
  * Added test-specific step limits for controlling test-mode duration
  * Enhanced dataloader with pinned memory support for improved GPU transfer efficiency
  * Added summary utility methods for tracking metrics and finding best values

* **Bug Fixes**
  * Improved dataset download reliability with URL fallback to alternate sources

* **Performance Improvements**
  * Optimized GPU device transfers and model state management

* **Documentation**
  * Updated for PyTorch-exclusive support in FastEstimator 2.0

### CODE REVIEW

File: fastestimator/estimator.py [hunk 1]
```diff
@@ -86,6 +88,7 @@ def __init__(self,
                  epochs: int,
                  train_steps_per_epoch: Optional[int] = None,
                  eval_steps_per_epoch: Optional[int] = None,
+                 test_steps_per_epoch: Optional[int] = None,
```
Comment (vbvg2008) [inline fastestimator/estimator.py:91]: seems like I was summoned to review this PR :)

haven't gone through the whole thing yet, but out of curiosity, what use case are we doing for `test_steps_per_epoch`? I thought test data is something we should just do only ONCE after all training is done, otherwise we have a risk of contaminating the test data.  The moment we use test data for any training decision (such as early stopping or best model saving), the test data becomes eval data. :)

just thoughts, no strong preference.

**Comment by vbvg2008** [comment]
also seems like there are great number of changes that call for a version bump

**Comment by aashokvardhan** [comment]
So dev branch was a development branch once these changes are pushed we will be merged to master to create 2.0 Fastestimator from current 1.7

File: fastestimator/estimator.py (hunk 1 above)
Comment (aashokvardhan) [reply fastestimator/estimator.py:91]: The test_steps_per_epoch is just used for consistency purposes, we aren't not using it for early stopping or any other training specific decisions.

File: fastestimator/dataset/data/medmnist.py [hunk 2]
```diff
@@ -19,29 +19,38 @@
...
+    "adrenalmnist3d": (f"{_ZENODO_BASE}/adrenalmnist3d.npz?download=1", "1E__zT_PqXlaiyW3r6Ii98UVVv5-nDZ_q"),
+    "bloodmnist": (f"{_ZENODO_BASE}/bloodmnist.npz?download=1", "17r9NENUnUoqnWZ-zJaveLU40R746A-ul"),
+    "breastmnist": (f"{_ZENODO_BASE}/breastmnist.npz?download=1", "1E3fWybVlOXSIjg_kFSQ-nwOikLgrbsJc"),
+    "chestmnist": (f"{_ZENODO_BASE}/chestmnist.npz?download=1", "1lGZtFjlviRf_vwmxB4NgApjCdfDuh8pk"),
+    "dermamnist": (f"{_ZENODO_BASE}/dermamnist.npz?download=1", "1k-vw5otV5rCQLVQKyxmo1mhI3KqDbwU5"),
+    "fracturemnist3d": (f"{_ZENODO_BASE}/fracturemnist3d.npz?download=1", "104OYiONKAsN2O-VeVGS7ReXZI4GtNsZk"),
+    "nodulemnist3d": (f"{_ZENODO_BASE}/nodulemnist3d.npz?download=1", "1krf_Uv-bA5CEt_dCD6h8xJbzZk9ETWLU"),
+    "octmnist": (f"{_ZENODO_BASE}/octmnist.npz?download=1", "1yRhEBDQwHu5jSnrzIm87Cazns3Eesv4b"),
+    # NOTE: no verified Google Drive fallback ID exists for organamnist; None disables the fallback
+    # to prevent the silent data-corruption that occurred when it shared octmnist's Drive ID.
```
Comment (DwijayDS) [inline fastestimator/dataset/data/medmnist.py]: Can we reupload organamnist and have that drive ID here?

File: fastestimator/dataset/data/medmnist.py [hunk 3]
```diff
@@ -100,7 +109,11 @@ def load_data(
     download_path = os.path.join(root_dir, f"{dataset_name}.npz")
 
     print("Downloading data to {}".format(root_dir))
-    download_file_from_google_drive(dataset_ids[dataset_name], download_path)
+    url, gdrive_id = dataset_sources[dataset_name]
+    if gdrive_id is not None:
+        download_url_with_fallback(url, gdrive_id, download_path)
+    else:
+        download_url(url, download_path)
```
Comment (DwijayDS) [inline fastestimator/dataset/data/medmnist.py]: Can we include gdrive_id being None in the `download_url_with_fallback` method. That way we only need to check in the case url doesnt work and we have to fallback. We can throw a warning in the case gdrive_id is None

File: fastestimator/dataset/data/medmnist.py [hunk 4]
```diff
@@ -19,29 +19,36 @@
...
+    "adrenalmnist3d": (f"{_ZENODO_BASE}/adrenalmnist3d.npz?download=1", "1E__zT_PqXlaiyW3r6Ii98UVVv5-nDZ_q"),
+    "bloodmnist": (f"{_ZENODO_BASE}/bloodmnist.npz?download=1", "17r9NENUnUoqnWZ-zJaveLU40R746A-ul"),
+    "breastmnist": (f"{_ZENODO_BASE}/breastmnist.npz?download=1", "1E3fWybVlOXSIjg_kFSQ-nwOikLgrbsJc"),
+    "chestmnist": (f"{_ZENODO_BASE}/chestmnist.npz?download=1", "1lGZtFjlviRf_vwmxB4NgApjCdfDuh8pk"),
+    "dermamnist": (f"{_ZENODO_BASE}/dermamnist.npz?download=1", "1k-vw5otV5rCQLVQKyxmo1mhI3KqDbwU5"),
+    "fracturemnist3d": (f"{_ZENODO_BASE}/fracturemnist3d.npz?download=1", "104OYiONKAsN2O-VeVGS7ReXZI4GtNsZk"),
+    "nodulemnist3d": (f"{_ZENODO_BASE}/nodulemnist3d.npz?download=1", "1krf_Uv-bA5CEt_dCD6h8xJbzZk9ETWLU"),
+    "octmnist": (f"{_ZENODO_BASE}/octmnist.npz?download=1", "1yRhEBDQwHu5jSnrzIm87Cazns3Eesv4b"),
+    "organamnist": (f"{_ZENODO_BASE}/organamnist.npz?download=1", "1yRhEBDQwHu5jSnrzIm87Cazns3Eesv4b"),
+    "organcmnist": (f"{_ZENODO_BASE}/organcmnist.npz?download=1", "1GqXyaV6arJq0O-3Fn2hvBw77MG9fHeoz"),
```
[the remark this answers is not in the document]
Comment (chm123) [reply fastestimator/dataset/data/medmnist.py:38]: @aashokvardhan You can remove duplicate in line 39.

File: tutorial/advanced/t15_finetuning.ipynb
Comment (chm123) [inline tutorial/advanced/t15_finetuning.ipynb:25]: This block should be removed. It's a duplicate the the PyTorch block below.

**Review by chm123 — APPROVED** [review]
